quinn/src/runtime/async_std.rs

```diff
@@ -23,10 +24,11 @@ impl Runtime for AsyncStdRuntime {
         async_std::task::spawn(future);
     }
 
-    fn wrap_udp_socket(&self, sock: std::net::UdpSocket) -> io::Result<Box<dyn AsyncUdpSocket>> {
+    fn wrap_udp_socket(&self, sock: std::net::UdpSocket) -> io::Result<Arc<dyn AsyncUdpSocket>> {
         udp::UdpSocketState::configure((&sock).into())?;
-        Ok(Box::new(UdpSocket {
+        Ok(Arc::new(UdpSocket {
             io: Async::new(sock)?,
+            state: udp::UdpState::new(),
```

Comment:
So, should we merge `udp::UdpState` and `udp::UdpSocketState` as well?

Reply:
Turned out great; added to the PR.